packages/core/src/app/customer/Customer.tsx

```diff
@@ -420,11 +423,25 @@ class Customer extends Component<CustomerProps & WithCheckoutCustomerProps & Ana
     private handleSignIn: (credentials: CustomerCredentials) => Promise<void> = async (
         credentials,
     ) => {
-        const { signIn, onSignIn = noop, onSignInError = noop } = this.props;
+        const {
+            executePaymentMethodCheckout,
+            signIn,
+            onSignIn = noop,
+            onSignInError = noop,
+            providerWithCustomCheckout,
+        } = this.props;
 
         try {
             await signIn(credentials);
-            onSignIn();
+
+            if (providerWithCustomCheckout === PaymentMethodId.Braintree || providerWithCustomCheckout === PaymentMethodId.BraintreeAcceleratedCheckout) {
+                await executePaymentMethodCheckout({
+                    methodId: providerWithCustomCheckout,
+                    continueWithCheckoutCallback: onSignIn,
+                });
+            } else {
+                onSignIn();
```

Comment:
We designed the `onSignIn` prop so that it can be replaced by other methods/requirements.
Can you please remove the `if` here and provide a proper value for `onSignIn`?

Reply:
I don't think we should replace this method with something else, because `onSignIn` method is a part of default flow that responsible of changing the checkout step (go to the next step). So this if statement saves default flow for cases, if we should not call execution method.

Also if we need to call execution method, onSignIn will be triggered under the hood, when the main execution part will be completed on checkout-sdk side.

Reply:
Can you please make a note of this so we can come back to think how we can work in a way that we don't need to add conditions.